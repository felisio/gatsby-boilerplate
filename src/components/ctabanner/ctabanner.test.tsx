import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import React from 'react';
import Ctabanner, { CtabannerProps } from './ctabanner';

const createTestProps = (props: CtabannerProps) => ({
    ...props,
});

const ctaBannerProps = createTestProps({
    title: 'Ready to jump start your career?',
    subtitle: 'Start learning with Quantic.',
    buttonCta: 'Try a lesson',
    buttonDestination: 'http://quantic.edu',
});

test('should verify Ctabanner Component props', () => {
    render(<Ctabanner {...ctaBannerProps} />);

    expect(screen.getByRole('heading')).toHaveTextContent(ctaBannerProps.title);
    expect(screen.getByText(ctaBannerProps.subtitle)).toHaveTextContent(ctaBannerProps.subtitle);
    expect(screen.getByRole('link')).toHaveTextContent(ctaBannerProps.buttonCta);
    expect(screen.getByRole('link')).toHaveAttribute('href', ctaBannerProps.buttonDestination);
});

test('should verify Ctabanner Component snapshot', () => {
    const tree = renderer.create(<Ctabanner {...ctaBannerProps} />).toJSON();

    expect(tree).toMatchSnapshot();
});
