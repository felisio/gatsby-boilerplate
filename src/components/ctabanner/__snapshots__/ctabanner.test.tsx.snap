// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should verify Ctabanner Component snapshot 1`] = `
.c2 {
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c3 {
  --standard-box-shadow: 0 1px 4px 0 var(--black-transparent-8),0 4px 6px 0 rgba(92,92,92,0.12);
  --white-on-blue-hover-box-shadow: 0 8px 12px 0 rgba(233,253,216,0.02),0 3px 8px 0 rgba(92,92,92,0.12);
  --watermelon-on-white-box-shadow: 0 1px 4px 0 rgba(0,0,0,0.08),0 4px 6px 0 rgba(92,92,92,0.12);
  --watermelon-on-white-hover-box-shadow: 0 8px 12px 0 rgba(233,253,216,0.02), 0 3px 8px 0 var(--black-transparent-12);
  --white-on-watermelon-box-shadow: 0 1px 4px 0 rgba(0,0,0,0.08),0 4px 6px 0 rgba(92,92,92,0.12);
  --white-on-watermelon-hover-box-shadow: 0 8px 12px 0 rgba(233,253,216,0.02),0 3px 8px 0 rgba(0,0,0,0.12);
  display: inline-block;
  width: auto;
  max-width: 28.8rem;
  border-radius: 4px;
  box-shadow: var(--standard-box-shadow);
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  text-align: center;
  outline-style: none;
}

.c3.medium {
  font-size: 1.3rem;
  line-height: 1.23;
  -webkit-letter-spacing: 0.6px;
  -moz-letter-spacing: 0.6px;
  -ms-letter-spacing: 0.6px;
  letter-spacing: 0.6px;
  padding: 1.2rem 2rem;
}

.c3.large {
  font-size: 1.5rem;
  line-height: normal;
  -webkit-letter-spacing: 0.9px;
  -moz-letter-spacing: 0.9px;
  -ms-letter-spacing: 0.9px;
  letter-spacing: 0.9px;
  padding: 1.5rem 2rem;
}

.c3.white-on-blue {
  color: var(--white);
  background-color: var(--dodger-blue);
}

.c3.white-on-blue:hover,
.c3.white-on-blue::selection {
  background-color: var(--dodger-blue-light);
  box-shadow: var(--white-on-blue-hover-box-shadow);
}

.c3.white-on-watermelon {
  color: var(--white);
  background-color: var(--wild-watermelon);
  box-shadow: var(--white-on-watermelon-box-shadow);
}

.c3.white-on-watermelon:hover,
.c3.white-on-watermelon::selection {
  background-color: var(--wild-watermelon-light);
  box-shadow: var(--white-on-watermelon-hover-box-shadow);
}

.c3.watermelon-on-white {
  color: var(--wild-watermelon);
  background-color: var(--white);
  box-shadow: var(--watermelon-on-white-box-shadow);
}

.c3.watermelon-on-white:hover,
.c3.watermelon-on-white::selection {
  opacity: 0.95;
  box-shadow: var(--watermelon-on-white-hover-box-shadow);
}

.c0 {
  position: relative;
  display: block;
  width: 100%;
  height: 346px;
  margin: 0;
}

.c0 .background {
  --shape-right: 11%;
  --shape-left: 100%;
  position: absolute;
  width: 100%;
  height: 100%;
  -webkit-clip-path: polygon(0 0,100% var(--shape-right),100% 100%,0% var(--shape-left));
  clip-path: polygon(0 0,100% var(--shape-right),100% 100%,0% var(--shape-left));
  background-image: linear-gradient(159deg,var(--pink-rose),var(--wild-watermelon));
}

.c0 .foreground {
  --foreground-top: 12px;
  --foreground-shape-right: 88%;
  position: absolute;
  top: var(--foreground-top);
  width: 100%;
  height: 332px;
  -webkit-clip-path: polygon(0 0,100% 0,100% 100%,0 var(--foreground-shape-right));
  clip-path: polygon(0 0,100% 0,100% 100%,0 var(--foreground-shape-right));
  background-image: linear-gradient(to right,rgba(170,125,238,0),rgba(170,125,238,0.15));
}

.c1 {
  --padding-top: 1.4rem;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
  height: 100%;
  padding: var(--padding-top) 2.5rem 0 2.5rem;
  color: var(--white);
  text-align: center;
}

.c1 div:first-child {
  margin-bottom: 3.5rem;
}

.c1 div:last-child a {
  width: 204px;
}

.c1 h2 {
  --margin-bottom: 1.3rem;
  margin: 0 0 var(--margin-bottom) 0;
  font-size: 3.2rem;
  font-weight: normal;
  font-stretch: normal;
  line-height: 1.19;
  color: var(--white);
  -webkit-letter-spacing: 0.2px;
  -moz-letter-spacing: 0.2px;
  -ms-letter-spacing: 0.2px;
  letter-spacing: 0.2px;
}

.c1 p {
  padding: 0;
  margin: 0;
  font-size: 1.8rem;
  font-weight: normal;
  font-stretch: normal;
  line-height: 1.44;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  opacity: 0.7;
}

@media (min-width:992px) {
  .c0 {
    height: 218px;
  }
}

@media (min-width:768px) {
  .c0 .background {
    --shape-right: 10%;
  }
}

@media (min-width:992px) {
  .c0 .background {
    height: 218px;
    --shape-right: 19%;
    --shape-left: 91%;
    background-image: linear-gradient(83deg,#e9314e 2%,#fc6e7f 47%,var(--wild-watermelon) 95%);
  }
}

@media (min-width:992px) {
  .c0 .foreground {
    height: 204px;
    --foreground-top: 15px;
    --foreground-shape-right: 76%;
    background-image: linear-gradient(to left,rgba(170,125,238,0.15),rgba(170,125,238,0));
  }
}

@media (min-width:768px) {
  .c1 {
    --padding-top: 1.9rem;
  }
}

@media (min-width:992px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    --padding-top: 5.1rem;
    text-align: left;
  }
}

@media (min-width:992px) {
  .c1 div:first-child {
    padding-right: 12rem;
    margin-bottom: 4.2rem;
  }
}

@media (min-width:992px) {
  .c1 div:last-child {
    -webkit-align-self: center;
    -ms-flex-item-align: center;
    align-self: center;
    padding: 0 0 3.2rem 0.5rem;
  }
}

@media (min-width:992px) {
  .c1 h2 {
    line-height: 1.31;
    --margin-bottom: 0.5rem;
  }
}

@media (min-width:992px) {
  .c1 p {
    font-size: 2.4rem;
    line-height: normal;
    -webkit-letter-spacing: 0.1px;
    -moz-letter-spacing: 0.1px;
    -ms-letter-spacing: 0.1px;
    letter-spacing: 0.1px;
  }
}

<section
  className="c0"
>
  <div
    className="background"
  />
  <div
    className="foreground"
  />
  <div
    className="c1"
  >
    <div>
      <h2>
        Ready to jump start your career?
      </h2>
      <p>
        Start learning with Quantic.
      </p>
    </div>
    <div>
      <a
        className="c2 c3 white-on-blue large"
        href="http://quantic.edu"
        target="_blank"
      >
        Try a lesson
      </a>
    </div>
  </div>
</section>
`;
